frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/ResetPasswordRequestHandler.java

```diff
@@ -317,23 +364,40 @@ yield generateApiGatewayProxyErrorResponse(
     private Optional<ErrorResponse> hasUserExceededMaxAllowedRequests(
             String email, UserContext userContext) {
         LOG.info("Validating Password Reset Count");
-        var codeRequestType =
-                CodeRequestType.getCodeRequestType(
-                        RESET_PASSWORD_WITH_CODE, JourneyType.PASSWORD_RESET);
-        var codeRequestCount = userContext.getAuthSession().getPasswordResetCount();
-        var codeRequestBlockedKeyPrefix = CODE_REQUEST_BLOCKED_KEY_PREFIX + codeRequestType;
-        var codeAttemptsBlockedKeyPrefix = CODE_BLOCKED_KEY_PREFIX + codeRequestType;
-        if (codeRequestCount >= configurationService.getCodeMaxRetries()) {
-            return Optional.of(ErrorResponse.TOO_MANY_PW_RESET_REQUESTS);
-        }
-        if (codeStorageService.isBlockedForEmail(email, codeRequestBlockedKeyPrefix)) {
-            LOG.info("Code is blocked for email as user has requested too many OTPs");
-            return Optional.of(ErrorResponse.BLOCKED_FOR_PW_RESET_REQUEST);
+        var userPermissionContext =
+                new UserPermissionContext(
+                        userContext.getAuthSession().getInternalCommonSubjectId(),
+                        null,
+                        email,
+                        userContext.getAuthSession());
+
+        var canSendResult =
+                permissionDecisionManager.canSendEmailOtpNotification(
+                        JourneyType.PASSWORD_RESET, userPermissionContext);
+
+        if (canSendResult.isSuccess()
+                && canSendResult.getSuccess() instanceof Decision.TemporarilyLockedOut lockedOut) {
+            var codeRequestCount = lockedOut.attemptCount();
+            // If count is -1, it means Redis block exists (subsequent requests after limit reached)
```

Comment:
This is quite low level (which I guess is why it needs a comment to explain it).  Is there another way to represent it, or is it a temporary thing?